utils/ccdatofhir/tests/test_mapping_allergy_intolerance.bal

```diff
@@ -0,0 +1,101 @@
+// in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+import ballerinax/health.fhir.r4;
+import ballerinax/health.fhir.r4.uscore501;
+
+@test:Config {}
+function testCcdaDocumentToFhirAllergyIntolerance() returns error? {
+    r4:Bundle|r4:FHIRError transformed = ccdaToFhir(ccdaDocumentValid);
+    test:assertTrue(transformed is r4:Bundle, "Error occurred while transforming CCDA document to FHIR!");
+
+    if (transformed is r4:Bundle) {
+        json jsonBundle = transformed.toJson();
+        
+        json[] entries = <json[]>check jsonBundle.entry;
+        json 'resource = check entries[2].'resource;
+        test:assertEquals(entries[2].'resource.resourceType, "AllergyIntolerance", "Incorrect resource type from the conversion!");
+        json[] clinicalStatusCoding = <json[]>check 'resource.clinicalStatus.coding;
+        test:assertEquals(clinicalStatusCoding[0].code, "active", "Incorrect clinical status from the conversion!");
+        test:assertEquals(clinicalStatusCoding[0].system, "http://terminology.hl7.org/CodeSystem/allergyintolerance-clinical", "Incorrect clinical status system from the conversion!");
+        json[] AllergyIntoleranceIdentifier = <json[]>check 'resource.identifier;
+        test:assertEquals(AllergyIntoleranceIdentifier.length(), 1, "Incorrect number of identifiers from the conversion!");
+        test:assertEquals(check AllergyIntoleranceIdentifier[0].system, "urn:ietf:rfc:3986", "Incorrect identifier system from the conversion!");
+        test:assertEquals(check AllergyIntoleranceIdentifier[0].value, "urn:oid:36e3e930-7b14-11db-9fe1-0800200c9a66", "Incorrect identifier value from the conversion!");
+        test:assertEquals(entries[2].'resource.onsetDateTime, "2023-05-31T22:05-05:00", "Incorrect onsetDateTime from the conversion!");
+        json[] allergyIntoleranceCodeCodings = <json[]>check 'resource.code.coding;
+        test:assertEquals(allergyIntoleranceCodeCodings.length(), 1, "Incorrect number of codes from the conversion!");
+        test:assertEquals(check allergyIntoleranceCodeCodings[0].system, "http://snomed.info/sct", "Incorrect system from the conversion!");
+        test:assertEquals(check allergyIntoleranceCodeCodings[0].code, "429625007", "Incorrect code from the conversion!");
+        test:assertEquals(check allergyIntoleranceCodeCodings[0].display, "Substance", "Incorrect display from the conversion!");
+        test:assertEquals(entries[2].'resource.recordedDate, "2014-10-03T10:30-05:00", "Incorrect recorded date from the conversion!");
+    }
```

Comment:
Let's have an else as well and assertfail if the result is an error. Otherwise even if the test fails due to a transformation error, we cannot catch.

Reply:
Updated in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/ef305c5f20ae58097861ee89041dde704d3d6ca4